cli/exp_mcp.go

```diff
@@ -106,12 +109,98 @@ func (*RootCmd) mcpConfigureClaudeDesktop() *serpent.Command {
+		testBinaryName   string
+	)
 	cmd := &serpent.Command{
-		Use:   "claude-code",
-		Short: "Configure the Claude Code server.",
-		Handler: func(_ *serpent.Invocation) error {
+		Use:   "claude-code <project-directory>",
+		Short: "Configure the Claude Code server. You will need to run this command for each project you want to use. Specify the project directory as the first argument.",
+		Handler: func(inv *serpent.Invocation) error {
+			if len(inv.Args) == 0 {
+				return xerrors.Errorf("project directory is required")
+			}
+			projectDirectory := inv.Args[0]
+			fs := afero.NewOsFs()
+			binPath, err := os.Executable()
+			if err != nil {
+				return xerrors.Errorf("failed to get executable path: %w", err)
+			}
+			if testBinaryName != "" {
+				binPath = testBinaryName
+			}
+			configureClaudeEnv := map[string]string{}
+			if _, ok := os.LookupEnv("CODER_AGENT_TOKEN"); ok {
+				configureClaudeEnv["CODER_AGENT_TOKEN"] = os.Getenv("CODER_AGENT_TOKEN")
+			}
+
+			if err := configureClaude(fs, ClaudeConfig{
+				AllowedTools:     []string{},
+				APIKey:           apiKey,
+				ConfigPath:       claudeConfigPath,
+				ProjectDirectory: projectDirectory,
+				MCPServers: map[string]ClaudeConfigMCP{
+					"coder": {
+						Command: binPath,
+						Args:    []string{"exp", "mcp", "server"},
+						Env:     configureClaudeEnv,
+					},
+				},
+			}); err != nil {
+				return xerrors.Errorf("failed to modify claude.json: %w", err)
+			}
+			cliui.Infof(inv.Stderr, "Wrote config to %s", claudeConfigPath)
+
+			// We also write the system prompt to the CLAUDE.md file.
+			if err := injectClaudeMD(fs, systemPrompt, claudeMDPath); err != nil {
+				return xerrors.Errorf("failed to modify CLAUDE.md: %w", err)
+			}
+			cliui.Infof(inv.Stderr, "Wrote CLAUDE.md to %s", claudeMDPath)
 			return nil
 		},
+		Options: []serpent.Option{
+			{
+				Name:        "claude-config-path",
+				Description: "The path to the Claude config file.",
+				Env:         "CODER_MCP_CLAUDE_CONFIG_PATH",
+				Flag:        "claude-config-path",
+				Value:       serpent.StringOf(&claudeConfigPath),
+				Default:     filepath.Join(os.Getenv("HOME"), ".claude.json"),
+			},
+			{
+				Name:        "claude-md-path",
+				Description: "The path to CLAUDE.md.",
+				Env:         "CODER_MCP_CLAUDE_MD_PATH",
+				Flag:        "claude-md-path",
+				Value:       serpent.StringOf(&claudeMDPath),
+				Default:     filepath.Join(os.Getenv("HOME"), ".claude", "CLAUDE.md"),
+			},
+			{
+				Name:        "api-key",
+				Description: "The API key to use for the Claude Code server.",
+				Env:         "CODER_MCP_CLAUDE_API_KEY",
+				Flag:        "claude-api-key",
+				Value:       serpent.StringOf(&apiKey),
+			},
+			{
+				Name:        "system-prompt",
+				Description: "The system prompt to use for the Claude Code server.",
+				Env:         "CODER_MCP_CLAUDE_SYSTEM_PROMPT",
+				Flag:        "claude-system-prompt",
+				Value:       serpent.StringOf(&systemPrompt),
+			},
```

Comment:
Would it make sense to set a default value here such as `Send a task status update that you are ready for input, and wait for user input`?